samples: cellular: modem_shell: Add support for counting pulses (#13739)
nrfconnect/sdk-nrf: samples/cellular/modem_shell/README.rst

     free:              160
     allocated:          88

----

GPIO pin pulse counter
======================

> You can use the ``gpio_count`` command to count pulses on a given GPIO pin.

Reviewer: We should probably add to the beginning and the first sentence modified to fit better:

MoSh command: ``gpio_count``

Author: I did not notice that most of the commands are documented like this. Not all are though, at least `sleep`, `uart` or `heap`. I will update all of these.

> When pulse counting has been enabled, LED 2 on nRF91 Series DKs shows the state of the pin input.

Reviewer 2: ```suggestion
When pulse counting has been enabled, **LED 2** on nRF91 Series DKs shows the state of the pin input.
```

> When pulse counting has been enabled, **LED 2** on nRF91 Series DKs shows the state of the pin input.

Reviewer: It seems there is a limitation that a single GPIO can be counted at a time. This should be mentioned somewhere.

.. note::

   The ``gpio_count enable`` command configures the GPIO pin as input and enables pull down.

.. code-block:: console

   mosh:~$ gpio_count get
   Number of pulses: 0
   mosh:~$ gpio_count enable 10
   mosh:~$ gpio_count get
   Number of pulses: 42

> 

Reviewer: Maybe add `gpio_count disable` command and get after that if it returns a value (I think it does based on code).

     free:              160
     allocated:          88

----

GPIO pin pulse counter
======================

MoSh command: ``gpio_count``

> You can use the command to count pulses on a given GPIO pin.

Reviewer 3: Please be more specific about how the counting actually happens. I am guessing you are counting rising edges?

Author: Good point, I'll add it. Yes, it is counting rising edges.

     free:              160
     allocated:          88

----

GPIO pin pulse counter
======================

MoSh command: ``gpio_count``

You can use the command to count pulses on a given GPIO pin.
A rising edge of the signal is counted as a pulse.
Pulse counting can be enabled only for a single pin at a time.

> When pulse counting has been enabled, **LED 2** on nRF91 Series DKs shows the state of the pin input.

Reviewer 2: ```suggestion
When pulse counting is enabled, **LED 2** on the nRF91 Series DKs shows the state of the pin input.
```
